internal/resource/config/localidentity/local_identity_identity_profiles_resource.go

```diff
@@ -0,0 +1,917 @@
+				resp.Diagnostics.AddError("Invalid Value for Attribute !", fmt.Sprintln("otp_length attribute must be a positive integer between 5 and 100."))
+			}
+			if internaltypes.IsDefined(emailVerificationConfig["require_verified_email_template_name"].(basetypes.StringValue)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("require_verified_email_template_name is not allowed when email verification or require_verified_email is disabled or when email_verification_type is OTP."))
+			}
+			if internaltypes.IsDefined(emailVerificationConfig["email_verification_sent_template_name"].(basetypes.StringValue)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("email_verification_sent_template_name is not allowed when email verification or require_verified_email is disabled or when email_verification_type is OTP."))
+			}
+		case "OTL":
+			if internaltypes.IsDefined(emailVerificationConfig["otp_length"].(basetypes.Int64Value)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("otp_length attribute is not allowed when email_verification_type is OTL. Required attribute: otl_time_to_live."))
+			}
+			if internaltypes.IsDefined(emailVerificationConfig["otp_retry_attempts"].(basetypes.Int64Value)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("otp_retry_attempts attribute is not allowed when email_verification_type is OTL. Required attribute: otl_time_to_live."))
+			}
+			if internaltypes.IsDefined(emailVerificationConfig["allowed_otp_character_set"].(basetypes.StringValue)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("allowed_otp_character_set attribute is not allowed when email_verification_type is OTL. Required attribute: otl_time_to_live."))
+			}
+			if internaltypes.IsDefined(emailVerificationConfig["otp_length"].(basetypes.Int64Value)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("otp_length attribute is not allowed when email_verification_type is OTL. Required attribute: otl_time_to_live."))
+			}
+			if internaltypes.IsDefined(emailVerificationConfig["email_verification_otp_template_name"].(basetypes.StringValue)) {
+				resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("email_verification_otp_template_name attribute is not allowed when email_verification_type is OTL. Required attribute: otl_time_to_live."))
+			}
+		}
+	}
+
+	if (internaltypes.IsDefined(model.ProfileEnabled) && (!model.ProfileEnabled.ValueBool())) && (internaltypes.IsDefined(model.RegistrationEnabled) && (!model.RegistrationEnabled.ValueBool())) {
+		resp.Diagnostics.AddError("Invalid Attribute Combination!", fmt.Sprintln("email, data_store_config, fields, registration_config and profile_config are not allowed when registration and profile are disabled."))
```

Comment:
It doesn't look to me like this if statement is checking what the error says. The if checks if profile_enabled is false and registration_enabled is false, but it doesn't check any of those other fields mentioned.